### ISSUE TITLE
Retransmission of confirmable messages are not working with LWIP

### ISSUE THREAD

**Issue by mbaril** [obgm/libcoap#59]
When coap_pdu is initialized, LWIP reserve space at the beginning of the pbuf to encode the data related to layers of the protocol stack that are under COAP (like UPD, IP, Ethernet) .

During transmission (udp_sendto), lwip encode the data of those layer and move the pbuf paylod pointer. However this pointer is not reset after the transmission.

Libcoap retries fail  because LWIP has not enought space to encode the UDP/IP headers.

In my case, I'm using Libcoap with ipv6 and I get the following error message from LWIP: "ip6_output: not enough room for IPv6 header in pbuf"

To fix this issue, I propose to reset the pbuf payload position after each transmission:

In method coap_send_impl (net.c):

```
  udp_sendto(context->endpoint->pcb, pdu->pbuf,
			ip6_2_ip(&dst->addr), dst->port);

  //Reset lwip pbuf to the state it was before the call to udp_sendto
  pdu->pbuf->payload = pdu->hdr;
```

Note: I'm not using the latest version of libcoap and I'm still using LWIP 1.4.1 but I think the problem is also present in latest version of the code. (to be validated)

**Comment by mrdeep1** [comment]
libcoap LwIP support has been re-written in PR https://github.com/obgm/libcoap/pull/884, and so this should no longer be an issue.

**Comment by mrdeep1** [comment]
Closed as fixed in #926 which is now in the develop branch.
